src/components/common/RedirectComponent/index.js

```diff
@@ -1,23 +1,47 @@
-import React from "react";
-import { Redirect } from "react-router";
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import { Redirect } from "react-router-dom";
 import { getQueryVariable } from "../../../common/utils";
-import Login from "../../../pages/Login";
-// import { PATHS } from "../../constant";
+import { METHODS } from "../../../services/api";
 import { PATHS } from "../../../constant";
 
 function RedirectComponent() {
-  const uri = window.location.href;
-  // const uri = `https://www.merakilearn.org/redirect?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZC[…]JKeaoO_2neSYUVf0fl1yjYlGt9YLNOeLr12Xs&redirectUrl=admission`;
+  const [emailId, setEmailId] = useState(null);
+  // const uri = window.location.href;
+  // const uri = `https://www.merakilearn.org/redirect?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjY0MSIsImVtYWlsIjoicG9vbmFtMTlAbmF2Z3VydWt1bC5vcmciLCJpYXQiOjE2NDU1MDgwNDIsImV4cCI6MTY3NzA2NTY0Mn0.az9n0hnUmj_2jHzj7LJ6ldAa_f74_gECF1BBDIzgU44&redirectUrl=admission`;
 
   const token = getQueryVariable("token");
   // undefined => "", remove leading slashes in redirect
   const redirect = (getQueryVariable("redirectUrl") || "").replace(/^\/+/g, "");
 
-  if (token) {
+  useEffect(() => {
+    axios({
+      method: METHODS.GET,
+      url: `${process.env.REACT_APP_MERAKI_URL}/users/me`,
+      headers: {
+        accept: "application/json",
+        Authorization: token,
+      },
+    }).then((res) => {
+      if (res.data.user.email) {
+        setEmailId(res.data.user.email);
+      }
+    });
+  }, []);
```

Comment:
As before, redux/saga would be better but if there's a time crunch here, to avoid code duplication, you could still use the API call from: https://github.com/navgurukul/bhanwari-devi/blob/6eef3ec78076e215e79e6d636f5510e73fbc4c2c/src/components/User/redux/api.js#L27-L36
```suggestion
  useEffect(() => {
    sendToken({token}).then((res) => {
      if (res.data.user.email) {
        setEmailId(res.data.user.email);
      }
    });
  }, []);
```
Of course, since you'd be revisiting this when you make the redux/saga change later, this may not make too much of a difference now.